string_pool.c

```diff
@@ -13,9 +13,10 @@ uint32_t string_pool_append(char *pool, uint32_t *n_pool, radixtree_t *r, const
         size_t n = strlen(str);
 
         /* TODO: check if n_pool + n < max */
-        strncpy (&pool[*n_pool], str, n);
```

Comment:
May I take this opportunity to plug OpenBSD's strlcpy function? It's nonstandard, but it was written specifically to prevent bugs like this one.